### PR TITLE
Improve bach docs layout

### PR DESCRIPTION
**Pull request by borft**

### CODE REVIEW

File: bach/docs/generate.py [hunk 1]
```diff
@@ -1,81 +1,250 @@
...
+from os import path, makedirs
 import json
-from typing import List, Dict, Any
-from lxml import html
-from lxml.etree import ElementTree
+from typing import Dict, List
+from lxml import html, etree
 
+"""
+Copyright Objectiv B.V. 2021
```
Comment (borft) [inline bach/docs/generate.py:9]: this is where the magic happens

File: bach/docs/source/conf.py [hunk 2]
```diff
@@ -18,13 +18,14 @@
 copyright = '2021, Objectiv'
 author = 'Objectiv B.V.'
 
+sys.path.extend([os.path.dirname(os.path.dirname(os.path.join(os.path.dirname(__file__))))])
 
-#ipython_execlines = [
+# ipython_execlines = [
```
Comment (jansentom) [inline bach/docs/source/conf.py:23]: plz remove

File: bach/docs/source/conf.py [hunk 3]
```diff
@@ -225,4 +230,4 @@ def autodoc_skip_member_bach_internal(app, what, name, obj, skip, options):
 def setup(app):
     app.connect("autodoc-process-docstring", remove_copyright_string)
     app.connect("autodoc-process-docstring", repair_classmethod_docstring)
-    app.connect("autodoc-skip-member", autodoc_skip_member_bach_internal)
+    app.connect("autodoc-skip-member", autodoc_skip_member_bach_internal)
```
Comment (jansentom) [inline bach/docs/source/conf.py:233]: gimme back my newline! :)

File: bach/docs/generate.py [hunk 4]
```diff
@@ -1,81 +1,250 @@
...
 
+# the actual index page, as `index.html` should contain the index, that is main navigation/ toctree
+index_page = 'intro'
+
+# these categories only get a toplevel entry in the sidebar
+# but no submenus for the headers
+skip_categories = ['Introduction', 'Core Concepts', 'Examples']
+
+# replacement label to avoid duplicates
+introduction = 'Overview'
```
Comment (jansentom) [inline bach/docs/generate.py:48]: yeah, maybe we need a better name for the variable.

File: bach/docs/generate.py [hunk 5]
```diff
@@ -1,81 +1,250 @@
...
+    :return: List[Dict] - can be directly json-dumped into a sidebar
+    """
+    sb = []
+    for menu_item in menu_items:
+        items = []
+        href = menu_item['href'].replace('.html', '')
+        label = menu_item['text']
+
+        # if this is a deeper nested item, and a method name, let's remove the classname
+        # for better readability.
```
Comment (jansentom) [inline bach/docs/generate.py:160]:
```suggestion
        # if this is a deeper nested item, and a fully qualified name, let's remove the class/module name
        # for a more compact tree
```

File: bach/docs/generate.py [hunk 6]
```diff
@@ -1,81 +1,250 @@
...
+        parts = label.split('.')
+        if level > 0 and len(parts) > 1:
+            label = parts[-1]
+
+        if '#' in href:
+            if menu_item['href'].startswith('intro'):
+                # intro is the "index" page, so correct path
+                href = menu_item['href'].replace('intro', '')
```
Comment (jansentom) [inline bach/docs/generate.py:168]:
```suggestion
            if menu_item['href'].startswith(index_page):
                # this is the "index" page, so correct path
                href = menu_item['href'].replace(index_page, '')
```

**Review by jansentom — APPROVED** [review]
Think this is quite good already. I did not review the mdx creation in detail, as that works and thats proof enough. Left some small comments

File: bach/docs/generate.py (hunk 4 above)
Comment (borft) [reply bach/docs/generate.py:48]: agreed, perhaps something like `category_index` or so

**Pull request merged**